### PR TITLE
:white_check_mark: add jsdoc linting

### PR DESCRIPTION
**Pull request by ianardee**
## Description
Clean and fix JSDoc annotations using [eslint-plugin-jsdoc](https://www.npmjs.com/package/eslint-plugin-jsdoc).

Remove `FieldConstructor` because it was a copy-paste of `BaseFieldConstructor`.

Only activated some of the linting options to keep compatibility with TSDoc.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires a change to the official Guide documentation.

### CODE REVIEW

File: src/input/base.ts [hunk 1]
```diff
@@ -50,12 +53,9 @@ export abstract class LocalInputSource extends InputSource {
...
 
   /**
-   * @param {String} inputType - the type of input used in file ("base64", "path", "dummy").
-   *                             NB: dummy is only used for tests purposes
-   * @param {Boolean} cutPages
-   * NB: Because of async calls, init() should be called after creating the object
+   * @param {InputConstructor} constructor Constructor parameters.
    */
-  constructor({ inputType }: InputProps) {
+  protected constructor({ inputType }: InputConstructor) {
```
Comment (sebastianMindee) [inline src/input/base.ts:58]: Is this really needed on abstract class?

File: src/parsing/standard/tax.ts [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+ * @property {string} constructor.rateKey - Key to use to get the tax rate in the prediction dict.
+ * @property {string} constructor.codeKey - Key to use to get the tax code in the prediction dict.
+ * @property {string} constructor.baseKey - Key to use to get the base tax in the prediction dict.
+ */
+interface TaxConstructor extends BaseFieldConstructor {
   rateKey?: string;
   codeKey?: string;
   baseKey?: string;
-  reconstructed?: boolean;
-  pageId?: number;
```
Comment (sebastianMindee) [inline src/parsing/standard/tax.ts:12]: Can't we have document-level taxes?

**Review by sebastianMindee — COMMENTED** [review]
Need to have a couple of points clarified, but otherwise :ok:

File: src/parsing/standard/tax.ts (hunk 2 above)
Comment (ianardee) [reply src/parsing/standard/tax.ts:12]: it inherits from the base params which have these properties

Comment (sebastianMindee) [reply src/parsing/standard/tax.ts:12]: Oh my bad I misread the diff cuz it cut off right below and started again with `pageId!: number;` so I thought you had enforced it, but it's only for the `TaxField` which makes more sense.

File: src/input/base.ts (hunk 1 above)
Comment (ianardee) [reply src/input/base.ts:58]: no, not required, just my IDE was nagging me about it

Comment (sebastianMindee) [reply src/input/base.ts:58]: Yeah fair enough

**Review by sebastianMindee — APPROVED** [review]

**Pull request merged**